docs/usage/validators.md

```diff
@@ -419,15 +292,120 @@ except ValidationError as e:
+If any field fails to validate after validators will not be called.
+
+## Handling errors in validators
 
-Field validation will not occur if `pre=True` root validators raise an error. As with field validators,
-"post" (i.e. `pre=False`) root validators by default will be called even if prior validators fail; this
-behaviour can be changed by setting the `skip_on_failure=True` keyword argument to the validator.
-The `values` argument will be a dict containing the values which passed field validation and
-field defaults where applicable.
+As mentioned in the previous sections validators can raise either a `ValueError` or `AssertionError` (including ones generated by `assert ...` statements).
+You can also raise a `PydanticCustomError` which is a bit more verbose but gives you extra flexibility.
+Any other errors (including `TypeError`) are bubbled up and not wrapped in a `ValidationError`.
+
+```python
+from pydantic_core import PydanticCustomError
+
+from pydantic import BaseModel, ValidationError, field_validator
+
+
+class Model(BaseModel):
+    x: int
+
+    @field_validator('x')
+    @classmethod
+    def validate_x(cls, v: int) -> int:
+        if v % 42 == 0:
+            raise PydanticCustomError(
+                'the_answer_error',
+                '{number} is the answer!',
+                {'number': v},
+            )
+        return v
+
+
+try:
+    Model(x=42 * 2)
+except ValidationError as e:
+    print(e)
+    """
+    1 validation error for Model
+    x
+      84 is the answer! [type=the_answer_error, input_value=84, input_type=int]
+    """
+```
+
+## Special Types
+
+Pydantic provides a few special types that can be used to customize validation.
+
+- [`InstanceOf`][pydantic.functional_validators.InstanceOf] is a type that can be used to validate that a value is an instance of a given class.
+
+```py
+from typing import List
+
+from pydantic import BaseModel, InstanceOf, ValidationError
+
+
+class Fruit:
+    def __repr__(self):
+        return self.__class__.__name__
+
+
+class Banana(Fruit):
+    ...
+
+
+class Apple(Fruit):
+    ...
+
+
+class Basket(BaseModel):
+    fruits: List[InstanceOf[Fruit]]
+
+
+print(Basket(fruits=[Banana(), Apple()]))
+#> fruits=[Banana, Apple]
+try:
+    Basket(fruits=[Banana(), 'Apple'])
+except ValidationError as e:
+    print(e)
+    """
+    1 validation error for Basket
+    fruits.1
+      Input should be an instance of Fruit [type=is_instance_of, input_value='Apple', input_type=str]
+    """
+```
+
+- [`SkipValidation`][pydantic.functional_validators.SkipValidation] is a type that can be used to skip validation on a field.
+
+```py
+from typing import List
+
+from pydantic import BaseModel, SkipValidation
+
+
+class Model(BaseModel):
+    names: List[SkipValidation[str]]
+
+
+m = Model(names=['foo', 'bar'])
+print(m)
+#> names=['foo', 'bar']
+
+m = Model(names=['foo', 123])  # (1)!
+print(m)
+#> names=['foo', 123]
+```
+
+1. Note that the validation of the second item is skipped.
```

Comment:
I'd maybe add a note that if the value has the wrong type due to skipped validation, a warning may be produced during serialization.